beacon-chain/state/stategen/hot.go

```diff
@@ -139,3 +132,34 @@ func (s *State) loadHotStateBySlot(ctx context.Context, slot uint64) (*state.Bea
 
 	return s.ReplayBlocks(ctx, startState, replayBlks, slot)
 }
+
+// This returns the last saved in DB ancestor state of the input block root.
+// It recursively look up block's parent until a corresponding state of the block root
+// is found in the DB.
+func (s *State) lastAncestorState(ctx context.Context, root [32]byte) (*state.BeaconState, error) {
+	ctx, span := trace.StartSpan(ctx, "stateGen.lastAncestorState")
+	defer span.End()
+
+	b, err := s.beaconDB.Block(ctx, root)
+	if err != nil {
+		return nil, err
+	}
+	if b == nil {
+		return nil, errUnknownBlock
+	}
+
+	for {
+		parentRoot := bytesutil.ToBytes32(b.Block.ParentRoot)
```

Comment:
Please check context deadline is not exceeded in this loop

Reply:
Right... Adding
```
if ctx.Err() != nil {
			return nil, ctx.Err()
		}
```